scheduler/src/cook/rate_limit/generic.clj

```diff
@@ -90,21 +109,28 @@
     [_]
     enforce?))
 
+(defn ^RateLimiter make-generic-token-bucket-filter
+  [{:keys [expire-minutes enforce?] :as config} make-tbf-fn]
+  {:pre [(> expire-minutes 0)
+         (or (true? enforce?) (false? enforce?))]}
+
+  (->TokenBucketFilterRateLimiter config
+                                  (-> (CacheBuilder/newBuilder)
+                                      (.expireAfterAccess expire-minutes (TimeUnit/MINUTES))
+                                      (.build (fn->CacheLoader make-tbf-fn)))
+                                  enforce?))
+
 (defn ^RateLimiter make-token-bucket-filter
-  [^long max-tokens ^double tokens-replenished-per-minute ^long bucket-expire-minutes enforce?]
-  {:pre [(> max-tokens 0)
-         (> tokens-replenished-per-minute 0)
-         (> bucket-expire-minutes 0)
+  [{:keys [bucket-size tokens-replenished-per-minute expire-minutes enforce?] :as config}]
+  {:pre [(> bucket-size 0)
+         (> tokens-replenished-per-minute 0.0)
```

Comment:
can (> tokens-replenished-per-minute 0.0) ever behave differently from (> tokens-replenished-per-minute 0)

Reply:
No. But this is a float to indicate that the argument is (expected) to be a float.